GUI/src/app/display-curve/display-curve.component.ts

```diff
@@ -0,0 +1,196 @@
+import { AfterViewInit, Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import functionPlot from 'function-plot';
+import { FormsModule } from "@angular/forms";
+import {
+    MatAccordion,
+    MatExpansionPanel,
+    MatExpansionPanelActionRow,
+    MatExpansionPanelDescription, MatExpansionPanelHeader, MatExpansionPanelTitle
+} from "@angular/material/expansion";
+import {MatButton} from "@angular/material/button";
+import {MatFormField, MatLabel} from "@angular/material/form-field";
+import {MatInput} from "@angular/material/input";
+
+@Component({
+    selector: 'app-display-curve',
+    standalone: true,
+    imports: [
+        FormsModule,
+        MatAccordion,
+        MatButton,
+        MatExpansionPanel,
+        MatExpansionPanelActionRow,
+        MatExpansionPanelDescription,
+        MatExpansionPanelHeader,
+        MatExpansionPanelTitle,
+        MatFormField,
+        MatInput,
+        MatLabel
+    ],
+    templateUrl: './display-curve.component.html',
+    styleUrls: ['./display-curve.component.scss']
+})
+export class DisplayCurveComponent implements OnInit {
+
+    @ViewChild('plotContainerAdd') plotContainerAdd!: ElementRef;
+    @ViewChild('plotContainerMul') plotContainerMul!: ElementRef;
+
+    // Curve for Addition
+    public a = -7;
+    public b = 10;
+    // Point on Curve
+    public P = { x: 1, y: 2 };
+    public Q = { x: 3, y: 4 };
+    // Result from point addition
+    public R = { x: 0, y: 0 };
+
+    // Curve for Multiplication
+    public a2 = -7;
+    public b2 = 10;
+    public n2 = 2;
+    public P2 = { x: 1, y: 2 };
+    // Result point from Multiplication
+    public Q2 = { x: -1, y: -4 };
+
+    constructor() { }
+
+    ngOnInit(): void { }
+
+    private checkIfPointIsOnCurve(x: number, y: number): boolean {
+        // y^2 = x^3 + ax + b
+        if (x == 0 && y == 0) {
+            return true; // point at infinity
+        }
+        return Math.pow(y, 2) === Math.pow(x, 3) + this.a * x + this.b;
+    }
+
+    private calculateYCoordinate(x: number): number {
+        return Math.sqrt(Math.pow(x, 3) + this.a * x + this.b);
+    }
+
+    private calculateAdditionPointR(p: { x: number, y: number }, q: { x: number, y: number }): { x: number, y: number } {
+        let slope: number;
+        let x3: number;
+        let y3: number;
+
+        // Check if P or Q is the point at infinity
+        if ((p.x == 0 && p.y == 0) || (q.x == 0 && q.y == 0)) {
+            return (p.x == 0 && p.y == 0) ? q : p;
+        }
+
+        // Check if the tangent at P is vertical
+        if (p.x == q.x && p.y == q.y) {
+            if (p.y == 0) {
+                return { x: 0, y: 0 };
```

Comment:
Folglich muss hier der unendliche Punkt zurückgegeben werden, nicht (0,0)

Reply:
Und müsste die Bedingung nicht eigentlich (p.x = q.x & p.y + q.y = 0) lauten? Bin mir gerade nicht ganz sicher.
So oder so fängst du den Fall aber mit deinem default-return am Ende der Methode ab. Sollte also trotzdem klappen

Reply:
Wie soll der Unendliche Punkt denn bitte aussehen? Null geht nicht, und etwas anderes sinnvolles sehe ich auch eher weniger. Also bitte beschreibe mal, wie du es dir vorstellst

Reply:
Nein die Formel muss laut Skript so sein wie in Zeile 82 + 83 :
Siehe Skript Seite 57 Tangentenverfahren: (x1=x2) && (y1 = y2 != 0)

Reply:
Ich sehe im Skript auch nicht, wo du das mit Curvenparameter b = 0 als Sonderbedingung steht